flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/InternalBacklogAwareTimerServiceImplTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.operators;
+
+import org.apache.flink.api.common.typeutils.base.IntSerializer;
+import org.apache.flink.api.common.typeutils.base.StringSerializer;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.runtime.state.KeyGroupRange;
+import org.apache.flink.runtime.state.KeyGroupedInternalPriorityQueue;
+import org.apache.flink.runtime.state.heap.HeapPriorityQueueSetFactory;
+import org.apache.flink.streaming.runtime.tasks.TestProcessingTimeService;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.streaming.api.operators.TimeServiceTestUtils.createBacklogTimerService;
+import static org.apache.flink.streaming.api.operators.TimeServiceTestUtils.createInternalTimerService;
+import static org.apache.flink.streaming.api.operators.TimeServiceTestUtils.createTimerQueue;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for {@link InternalBacklogAwareTimerServiceImpl}. */
+public class InternalBacklogAwareTimerServiceImplTest extends InternalTimerServiceImplTest {
```

Comment:
Let's override the way `InternalTimerServiceImplTest` uses to create timer service into producing `InternalBacklogAwareTimerServiceImpl` as well. Otherwise extending `InternalTimerServiceImplTest` will not test `InternalBacklogAwareTimerServiceImpl` against more test cases.